### ISSUE TITLE
Mapping an iterable function to string comes up with incompatible type error

### ISSUE THREAD

**Issue by huntfx** [python/mypy#11682]
**Bug Report**

I was needing to convert a list of strings to make each string a set. Instead of doing `(set(s) for s in lst)`, I did `map(set, lst)`.
Mypy warns that you are trying to map incompatible types, but it's perfectly valid Python since you can iterate over strings.

**To Reproduce**

1. Get a list of strings
2. Try use `map` to convert each string to a `set`/`list`/`tuple` etc

```python
a = ['a', 'bc']
b = map(list, a)  # [['a'], ['b', 'c']]
```

**Expected Behavior**

    Success: no issues found in 1 source file

**Actual Behavior**

    error: Argument 1 to "map" has incompatible type "Type[List[Any]]"; expected "Callable[[str], List[_T]]"

**Your Environment**

- Mypy version used: 0.910
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.7 and 3.9
- Operating system and version: Windows 10

**Comment by hauntsaninja** [comment]
Dumb workaround is `b = map(lambda x: list(x), a)` (or type-ignore)

**Comment by huntfx** [comment]
Found another potentially related issue, same environment:

```python
from typing import Dict, Iterable, List
values: List[str] = ['a']
replace: Dict[str, str] = {'a': 'b'}
x: Iterable[str] = map(replace.get, values)

```
    # error: Argument 1 to "map" has incompatible type overloaded function; expected "Callable[[str], str]"

**Comment by Jacob-Stevens-Haas** [comment]
Ran into (what I think) is this today with 
```python
max_or_min = ["max", "min"]
optfuns = [np.nanmax if opt == "max" else np.nanmin for opt in max_or_min]
zip(optfuns, results)
```
```
Argument 1 to "zip" has incompatible type "list[overloaded function]"; expected "Iterable[overloaded function]" 
```

**Comment by ilevkivskyi** [comment]
This now works correctly.

@huntfx two of your examples are legitimate errors because `dict.get()` can return `None`, other two are unrelated bug(s), please open a separate issue, if those use cases of `dict` constructor are important for you.
